### PR TITLE
[docs] Fixed errors while generating docs.

### PR DESCRIPTION
**Pull request by psachin**
I see below errors when generating docs using `make`. Decided to
cleanup a bit. The docstring look better than previous and no more
errors when generating docs.
```
/home/psachin/github/sos/sos/plugins/__init__.py:docstring of sos.plugins.Plugin.add_journal:5: ERROR: Unexpected indentation.
/home/psachin/github/sos/sos/plugins/__init__.py:docstring of sos.plugins.Plugin.add_journal:6: WARNING: Block quote ends without a blank line; unexpected unindent.
/home/psachin/github/sos/sos/plugins/__init__.py:docstring of sos.plugins.Plugin.add_journal:13: WARNING: Definition list ends without a blank line; unexpected unindent.
/home/psachin/github/sos/sos/plugins/__init__.py:docstring of sos.plugins.Plugin.add_journal:15: ERROR: Unexpected indentation.
/home/psachin/github/sos/sos/plugins/__init__.py:docstring of sos.plugins.Plugin.add_journal:16: WARNING: Block quote ends without a blank line; unexpected unindent.
/home/psachin/github/sos/sos/plugins/__init__.py:docstring of sos.plugins.Plugin.add_journal:18: WARNING: Definition list ends without a blank line; unexpected unindent.
```

Let me know if anything else needs cleanup in `sos/sos/plugins/__init__.py`

Signed-off-by: Sachin Patil <psachin@redhat.com>

---
Please place an 'X' inside each '[]' to confirm you adhere to our [Contributor Guidelines](https://github.com/sosreport/sos/wiki/Contribution-Guidelines)

- [X] Is the commit message split over multiple lines and hard-wrapped at 72 characters?
- [X] Is the subject and message clear and concise?
- [X] Does the subject start with **[plugin_name]** if submitting a plugin patch or a **[section_name]** if part of the core sosreport code?
- [X] Does the commit contain a **Signed-off-by: First Lastname <email@example.com>**?

### CODE REVIEW

**Comment by bmr-cymru** [comment]
Hi Sachin, thank you for working on this, unfortunately it isn't something we can merge in its current form. Although your changes are neater and include a lot of definite improvements (properly tagging code with `` for e.g.), we are using Sphinx for our documentation and I would prefer to have the documentation annotations in the native Sphinx format.

These look like:

```
        """Format BootParams as a string.

            Format this ``BootParams`` object as a string, with optional
            prefix, suffix, and value quoting.

            :param quote: A bool indicating whether to quote values.
            :param prefix: An optional prefix string to be concatenated
                           with the start of the formatted string.
            :param suffix: An optional suffix string to be concatenated
                           with the end of the formatted string.
            :returns: a formatted representation of this ``BootParams``.
            :returntype: string
        """
```

And mean that the generated Sphynx HTML docs have all the proper markup and CSS applied.

You can see how the above block formats [here](https://boom.readthedocs.io/en/latest/boom.html#module-boom.bootloader).

Improving both our docstring content and consistency has been a goal for a very long time and if you're happy to continue working on this I'd be pleased to review and merge your changes.

You can see many examples of this style of markup in the `boom` project (which documents almost every single submodule, member, function and class):

  https://github.com/bmr-cymru/boom/

**Comment by psachin** [comment]
@bmr-cymru 

I tried to keep the changes in Sphinx format. Not all functions are
changed in this patch but the one generating errors.

I'd like to continue improving the docs

**Comment by bmr-cymru** [comment]
Hi Sachin,

I see you've switched the patch to use the native Sphinx docstring format - that's the correct style to use for `sos`, thanks.

When I first looked at the patch the new docstrings were using the Google notation, e.g.:

```diff
-        """ Collect journald logs from one of more units.
-
-        Keyword arguments:
-        units     -- A string, or list of strings specifying the systemd
-                     units for which journal entries will be collected.
-        boot      -- A string selecting a boot index using the journalctl
-                     syntax. The special values 'this' and 'last' are also
-                     accepted.
-        since     -- A string representation of the start time for journal
-                     messages.
-        until     -- A string representation of the end time for journal
-                     messages.
-        lines     -- The maximum number of lines to be collected.
-        allfields -- Include all journal fields regardless of size or
-                     non-printable characters.
-        output    -- A journalctl output control string, for example
-                     "verbose".
-        timeout   -- An optional timeout in seconds.
+        """Collect journald logs from one of more units.
+
+        Attributes:
+
+            units (list): A string, or list of strings specifying the
+            systemd units for which journal entries will be collected.
+
+            boot (str): A string selecting a boot index using the
+            journalctl syntax. The special values 'this' and 'last'
+            are also accepted.
+
+            since (str): A string representation of the start time for
+            journal messages.
+
+            until (str): A string representation of the end time for
+            journal messages.
+
+            lines (int): The maximum number of lines to be collected.
+
+            allfields (bool): Include all journal fields regardless of
+            size or non-printable characters.
+
+            output (str): A journalctl output control string, for
+            example "verbose".
+
+            timeout (int): An optional timeout in seconds.
+
```

The patch does modify a number of other function and method comments: as a general note for future work, it's best to keep each commit to be one feature or fix (i.e. this patch could have been broken up to the major fix for the `add_journal()` problem, with the remaining whitespace and backtick markup in one or more following patches.

I'd be happy to review any further docs changes you have.

File: sos/plugins/__init__.py [hunk 1]
```diff
@@ -88,40 +88,43 @@ def _node_type(st):
...
-
-    packages (files) is an iterable of the names of packages (the paths
-    of files) to check for before running this plugin. If any of these packages
-    or files is found on the system, the default implementation of
-    check_enabled will return True.
-
-    profiles is an iterable of profile names that this plugin belongs to.
-    Whenever any of the profiles is selected on the command line the plugin
-    will be enabled (subject to normal check_enabled tests).
+    Attributes:
```
Comment (bmr-cymru) [inline sos/plugins/__init__.py]: Hi Sachin,

Unfortunately this class docstring is still using Google notation: *any* docstring that has "Attributes:" etc. labels is Google-style. The native Sphinx format uses RST.

Please [see here](https://thomas-cokelaer.info/tutorials/sphinx/_modules/template.html#MainClass1) for an example of a correct Sphinx-style class docstring using RST notation.

**Comment by bmr-cymru** [comment]
I'm going to cherry-pick the fix for the build errors from this branch - that way we can have those in the next release while you continue to work on the other changes.

**Comment by bmr-cymru** [comment]
I merged the fix for the `add_journal()` problems last year - if you're interested in pursuing any of the other changes in this PR please create a new pull request for those.

**Pull request closed without merging**